target/linux/ath79/base-files/etc/board.d/04_oem_config

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+#
+# Copyright (C) 2019 OpenWrt.org
+#
+
+. /lib/functions/uci-defaults.sh
+
+board_config_update
+
+board=$(board_name)
+
+case "$board" in
+devolo,magic-2-wifi)
+	ucidef_set_hostname devolo-$(grep SerialNumber /dev/mtd1 | grep -o ...$)
+	ucidef_set_wifi_ssid devolo-$(grep SerialNumber /dev/mtd1 | grep -o ...$)
```

Comment:
The problems I see with using the default vendor SSID and especially with using the serial number are that it introduces different behavior on different devices and the public serial number could potentially introduce security problems like a flaw which enables an attacker to guess the PSK which was demonstrated multiple times in the past few years. Thus OpenWrt should not trust these values to be independent from each other and choose BSSID/SSID in a way which does neither represent the real MAC address nor any value which could be a hint for an attacker to guess the PSK.

Reply:
I agree with your concern @CodeFetch and thank you for noticing it!
I see these solutions:

1. Create an extra flag that indicates if oem bssid is secure (not generated in a risky way like devolo is doing with serial number), this allows us to still have the oem value, but we should check this flag before using it to create a bssid with oem key... 

2. In the cases were the oem bssid is not secure, don't make the oem value available.

3. This can be used as "this" oem_bssid = { <original value>,<desired value for bssid> }, this allows us to collect the original value and always define an "original safe" bssid value, so if the original oem bssid is safe, both values should be the same, but if the original value is risky like in devolo case, provide a safe alternative. 
Example: oem_bssid =  { "devolo-123", "devolo-ABC"} where ABC it's the last 3 digits md5sum(psk), or { "devolo-123", "devolo-magic-2-wifi"}, or  { "devolo-123", "OpenWrt-devolo"}

I think the third option it's the best because provides oem value and a safe value to avoid psk generators...

Any thoughts on your side?

Reply:
You're mixing up SSID and BSSID. BSSID is the WLAN MAC address. Using the vendor SSID is rarely risky compared to the BSSID. But as I don't see any gain in keeping the vendor SSID I favor a generic "OpenWrt-" SSID with a random hex-digit-suffix, because it is consistent. Users don't expect the WLAN network to have the same SSID as default when they flash OpenWrt.